mantidproject/mantid
# Use consistent naming for spectra in Indirect Reduction

### Description of work
This PR fixes a bug where the reduced files coming out of an Indirect Reduction did not have consistent labeling for the spectra when comparing a summed run to non-summed runs. Additionally, using an "Individual" grouping instead of "All" or "Groups" would result in different spectra numbering for the first spectrum.

Fixes #37803

### To test:
**Test A**
1. Open Indirect Data Reduction
2. Choose TOSCA instrument
3. Load run number 30619-20
4. Click `Run`
5. Right click output workspaces and plot spectra. It says 0-2
6. Tick `Sum Files`
7. Click `Run`
8. Right click output workspaces and plot spectra. It says 0-2

**Test B**
1. Open Indirect Data Reduction
2. Choose IRIS instrument
3. Load run number 26176-77
4. Choose "Individual" detector grouping
5. Click `Run`
6. Right click output workspaces and plot spectra. The spectra numbering should start from 0
7. Choose "All" detector grouping
8. Click `Run`
9. Right click output workspaces and plot spectra. The spectra numbering should start from 0

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
